templates/person/show.json.twig

```diff
@@ -0,0 +1,94 @@
+            ,"gender": "{% include 'person/gender.html.twig' %}"
+            {% if (person.wikipediaUrl) or (person.viafUrl) %}
+            ,"sameAs": [
+                {% if person.wikipediaUrl %} "{{ person.wikipediaUrl}}"{% endif %}
+                {% if (person.wikipediaUrl) and (person.viafUrl) %} , {% endif %}
+                {% if person.viafUrl %} "{{ person.viafUrl}}" {% endif %}
+            ]
+            {% endif %}
+            {% if person.dob %},"birthDate": "{{person.dob}}" {% endif %}
+            {% if person.dod %}, "deathDate": "{{person.dod}}" {% endif %}
+            {% if person.cityOfBirth %},
+                "birthPlace": {
+                    "@type": "Place",
+                    "name": "{{person.cityOfBirth}}",
+                    "geo": {
+                        "@type": "GeoCoordinates",
+                        "latitude": "{{person.cityOfBirth.latitude}}",
+                        "longitude": "{{person.cityOfBirth.longitude}}"
+                    }
+                }
+             {% endif %}
+            {% if person.cityOfDeath %},
+                "deathPlace": {
+                    "@type": "Place",
+                    "name": "{{person.cityOfDeath}}",
+                    "geo": {
+                        "@type": "GeoCoordinates",
+                        "latitude": "{{person.cityOfDeath.latitude}}",
+                        "longitude": "{{person.cityOfDeath.longitude}}"
+                    }
+                }
+            {% endif %}
+            ,"@id": "{{ url('person_show', {'id':person.id}) }}"
+        }
+        {% for titleRole in person.getTitleRoles("Author") %}
+         {% if titleRole.Role.Name == "Author" %}
+                {% set book = titleRole.title %}
+                {% if book %}
+                ,{
+                    "@type": "Book",
+                    "@id": "{{ url('title_show', {'id':book.id}) }}" ,
+                    "name": "{{book.title}}"
+                    {% if book.genre %} ,"genre": "{{book.genre}}" {% endif %}
+                    {% if book.pubdate %},"datePublished": "{{book.pubdate}}" {% endif %}
+                    {% if book.locationOfPrinting %},
+                        "locationCreated": {
+                                    "@type": "Place",
+                                     "name": "{{book.locationOfPrinting}}",
+                                    "geo": {
+                                        "@type": "GeoCoordinates",
+                                        "latitude": "{{book.locationOfPrinting.latitude}}",
+                                        "longitude": "{{book.locationOfPrinting.longitude}}"
+                                    }
+                                }
+                    {% endif %}
+                     ,"author": [
+                        {% for titleRole in book.getTitleRoles("Author") %}{% set author = titleRole.person %}
```

Comment:
I see the inconsistency in the getTitleRoles/getTitleFirmRoles functions now. This one _does_ accept a string and will filter based on the string. That's some bad design on my part.

Reply:
I think maybe we can discuss this if you also agree? Just to let me get more ideas about this Title Roles again and why it works in some cases and why does not in some cases :D